src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,553 @@
+#include "MagnumPlugins/KtxImporter/KtxHeader.h"
+
+namespace Magnum { namespace Trade {
+
+namespace {
+
+std::size_t imageLength(const Vector3i& size, PixelFormat format) {
+    return size.product()*pixelSize(format);
+}
+
+/** @todo Use CompressedPixelStorage::dataProperties for this */
+std::size_t imageLength(const Vector3i& size, CompressedPixelFormat format) {
+    const Vector3i blockSize = compressedBlockSize(format);
+    const Vector3i blockCount = (size + (blockSize - Vector3i{1})) / blockSize;
+    return blockCount.product()*compressedBlockDataSize(format);
+}
+
+void endianSwap(Containers::ArrayView<char> data, UnsignedInt typeSize) {
+    switch(typeSize) {
+        case 1:
+            /* Single-byte or block-compressed format, nothing to do */
+            break;
+        case 2:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<UnsignedShort>(data));
+            break;
+        case 4:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<UnsignedInt>(data));
+            break;
+        case 8:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<UnsignedLong>(data));
+            break;
+        default:
+            CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+            break;
+    }
+}
+
+void swizzlePixels(const PixelFormat format, Containers::Array<char>& data, const char* verbosePrefix) {
+    if(format == PixelFormat::RGB8Unorm) {
+        if(verbosePrefix) Debug{} << verbosePrefix << "converting from BGR to RGB";
+        auto pixels = reinterpret_cast<Math::Vector3<UnsignedByte>*>(data.data());
+        std::transform(pixels, pixels + data.size()/sizeof(Math::Vector3<UnsignedByte>), pixels,
+            [](Math::Vector3<UnsignedByte> pixel) { return Math::gather<'b', 'g', 'r'>(pixel); });
+    } else if(format == PixelFormat::RGBA8Unorm) {
+        if(verbosePrefix) Debug{} << verbosePrefix << "converting from BGRA to RGBA";
+        auto pixels = reinterpret_cast<Math::Vector4<UnsignedByte>*>(data.data());
+        std::transform(pixels, pixels + data.size()/sizeof(Math::Vector4<UnsignedByte>), pixels,
+            [](Math::Vector4<UnsignedByte> pixel) { return Math::gather<'b', 'g', 'r', 'a'>(pixel); });
+
+    } else CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+}
+
+/** @todo Swizzle channels if necessary. Vk::PixelFormat doesn't contain any
+          of the swizzled formats like B8G8R8A8. */
+/** @todo Support all Vulkan formats allowed by the KTX spec. Create custom
+          PixelFormat with pixelFormatWrap and manually fill PixelStorage/
+          CompressedPixelStorage. We can take all the necessary info from
+          https://github.com/KhronosGroup/KTX-Specification/blob/master/formats.json
+          Do we also need this for the KtxImageConverter? This would allow
+          users to pass in images with implementation-specific PixelFormat
+          using the Vulkan format enum directly. */
```

Comment:
The DFD has pixel/block size, but I figured since we have to map this information for the converter anyway, we might reuse it as 'ground truth' and not even deal with the DFD at all, or if we choose to, have a way to validate the info.

Reply:
For code size reasons I'd rather not have the whole table in the importer, only in the converter. 

> have a way to validate the info

I'd definitely do that, but only for pixel / block size, not the other fields like channel bits and whatever else is there. Magnum has [pixelSize()](https://doc.magnum.graphics/magnum/namespaceMagnum.html#a0b7bf2313cad99f16b84cd0614938273) / [compressedBlockSize()](https://doc.magnum.graphics/magnum/namespaceMagnum.html#adb6a6c9d15a04272bbeaa4f42e552f3e) / [compressedBlockDataSize()](https://doc.magnum.graphics/magnum/namespaceMagnum.html#aeed2b6db91a62ffcb28752a178fa52ad) so I think it could be just checking these against the pixel format and relevant fields in the DFD and failing if these don't match?